content/blogs/2024-10-15-huggin-face-kestra-http.md

```diff
@@ -0,0 +1,102 @@
+
+ Simply connect to Hugging Face’s API through Kestra’s HTTP tasks, and you’re good to go. Kestra will handle API authentication, HTTP requests, and even responses.
+
+## How It Works: Connecting to Hugging Face Models
+
+The API calls work like any other HTTP task, allowing you to integrate AI features into your workflows. Here’s how you can make the most of Hugging Face models within Kestra for different purposes:
+
+1. **AI Data Insight**: Kestra can trigger Hugging Face models that analyze data in real-time, and give you some insights about the incoming data, is it a good day for your sales? What's your most requested products? You can even push further with some alerting to send a Slack/Discord message giving you the trends.
+2. **Sentiment Analysis for Customer Support**: Connect Kestra to your customer service channels and route incoming messages to Hugging Face’s sentiment analysis models. Kestra can classify the message tone and urgency, escalating high-priority feedback to the right teams instantly.
+3. **Language Translation**: If you need to manage multilingual customer inquiries, Kestra can automatically send incoming messages to a Hugging Face translation model, then respond in the customer’s language in seconds. It’s a quick way to offer native language support without extensive setup.
+
+### Implementing a Real Example in Kestra
+
+Let’s look at a practical example of using Kestra to translate text from English to Spanish. With Hugging Face’s NLP models, you can configure an HTTP task to make a simple API call. Here’s how it’s done in Kestra:
+
+```yaml
+id: hugging_face_translation
+namespace: company.team
+tasks:
+  - id: translate_text
+    type: io.kestra.plugin.core.http.Request
+    uri: https://api-inference.huggingface.co/models/Helsinki-NLP/opus-mt-en-es
+    method: POST
+    contentType: application/json
+    headers:
+      Authorization: "Bearer your token"
+    formData:
+      inputs: "Hello from Paris"
+```
+
+In this workflow, Kestra makes a POST request to the Hugging Face model. This task sends text to the translation API, which then returns the translated message. It’s a lightweight integration, avoiding any extra steps for setup or maintenance. You can embed this task within a larger workflow, using the output as needed.
+
+### Taking It a Step Further
+
+Because Kestra is event-driven, you can trigger Hugging Face models whenever a specific event occurs. For example, let’s look at how Kestra can help streamline customer support by classifying requests in real-time. Each time a customer inquiry comes in, Kestra can automatically call a Hugging Face model to categorize the inquiry based on topics like "refund," "legal," or "FAQ." With Kestra’s HTTP capabilities, you can make these calls and get instant feedback within a single, orchestrated workflow.
```

Comment:
@MartinRst this section is confusing, you keep saying real-time + event-driven but you don't use any event triggers or RealtimTriggers. @Ben8t maybe you can help Martin here?

Reply:
Yep it's the step further workflow ben is on it